### DISCUSSION TITLE
Any Method To Differentiate Between Keyboard Arrow Keys And Direction Pad Of Gamepad #1 ?

### DISCUSSION THREAD

**Discussion by ghost** [nesbox/TIC-80#2084] [Q&A] [answered]
Hi,
```
btn(0)
btn(1)
btn(2)
btn(3)
```
Above seems to work simultaneously for keyboard arrow keys and gamepad # 1 directional pad.
How would we differentiate between keyboard and gamepad # 1 ?

Let me know, thanks!

Jesse

**Comment by nesbox** [comment]
Hi,
use `key()` for keyboard https://github.com/nesbox/TIC-80/wiki/key
use `btn()` for gamepad https://github.com/nesbox/TIC-80/wiki/btn

**Comment by ghost** [comment]
Hi,

I tried that but still have below issue:
Both keyboard arrow keys and direction pad of gamepad # 1 trigger btn() events simultaneously?
Is there a method to disable a gamepad when it's not being used by the player?
Thanks!

Jesse

**Comment by nesbox** [comment]
Unfortunately, there is no method to do this, you can only go to the settings and clear gamepad mapping values.

**Comment by ghost** [comment]
Hi,

Ok, but this is a bug - can it be fixed in next release?

Jesse

**Comment by nesbox** [comment, marked as answer]
No, it's not a bug, it's by design, we use keyboard to allow users play a game without gamepad, we have special section in the settings for such purposes called `gamepad mapping`

**Comment by ghost** [comment]
Ok, thanks!
